src/internal.c

```diff
@@ -3329,33 +3333,55 @@ static int DoKexDhReply(WOLFSSH* ssh, byte* buf, word32 len, word32* idx)
                 ret = WS_INVALID_ALGO_ID;
 #endif
             }
-#ifndef WOLFSSH_NO_SABER_LEVEL1_SHA256
-            else if (ssh->handshake->useSaber) {
-                /* This is a KEM. In this case, I need to decapsulate the
-                 * ciphertext. */
-                OQS_KEM* kem = NULL;
-                ret = 0;
+#ifndef WOLFSSH_NO_ECDH_SHA2_NISTP256_KYBER_LEVEL1_SHA256
+            else if (ssh->handshake->useEccKyber) {
+                /* This is a a hybrid of ECDHE and a post-quantum KEM. In this
+                 * case, I need to generated the ECC shared secret and
+                 * decapsulate the ciphertext of the post-quantum KEM. */
+                OQS_KEM* kem = OQS_KEM_new(OQS_KEM_alg_kyber_512);
+                if (kem == NULL) {
+                    ret = WS_INVALID_ALGO_ID;
+                }
 
+                ret = wc_ecc_init(key_ptr);
+#ifdef HAVE_WC_ECC_SET_RNG
+                if (ret == WS_SUCCESS)
+                    ret = wc_ecc_set_rng(key_ptr, ssh->rng);
+#endif
                 if (ret == 0) {
-                    kem = OQS_KEM_new(OQS_KEM_alg_saber_lightsaber);
-                    if (kem == NULL) {
-                        ret = WS_INVALID_ALGO_ID;
-                    }
+                    ret = wc_ecc_import_x963(f, fSz -
+                              (word32)kem->length_ciphertext, key_ptr);
                 }
 
                 if (ret == 0) {
-                    if (fSz != kem->length_ciphertext) {
-                        ret = WS_BUFFER_E;
-                    }
+                #ifdef PRIVATE_KEY_UNLOCK
```

Comment:
You do not need to wrap `PRIVATE_KEY_LOCK` with `#ifdef`. The macro handles this.

Reply:
Are you sure?  This pattern appears 10 times in `internal.c` .

Reply:
I'll remove all instances.